packages/language/test/test-builder.ts

```diff
@@ -932,6 +938,61 @@ export class TestBuilder {
     });
   }
 
+  expectTypeAt(label: string, expectedType: TypeExpectation): void {
+    const ranges = this.getLabelRanges(label);
+    for (const [start] of ranges) {
+      const node = this.syntaxNodesByOffset.get(start);
+      if (!node) {
+        throw new Error(
+          `No syntax node found at position ${this.createPositionMessage(start)}`,
+        );
+      }
+      const actualType = this.unit.services.inferer.inferType(node, this.unit);
+      this.expectTypeWithStructure(expectedType, actualType);
+    }
+  }
+
+  private expectTypeWithStructure(
+    expectedType: TypeExpectation,
+    actualType: TypeDescriptions.Any,
+  ) {
+    if (expectedType.type === DataType.Structure) {
+      if (actualType.type !== DataType.Structure) {
+        throw new Error(
+          `Expected type to be a ${TypeDescriptions.Names[DataType.Structure]}, but got ${TypeDescriptions.Names[actualType.type]}`,
+        );
+      }
+      for (const [name, expectedMemberType] of Object.entries(
+        expectedType.members ?? {},
+      )) {
```

Comment:
Suggestion: This doesn't catch any additional properties that might be inside of `actualType`, but that aren't expected within `expectedType`. I.e. this will result in a false positive when we have something like `expectedType: {}`, `actualType: { name: String }`.